src/dstore/pmix_esh.c

```diff
@@ -576,33 +576,29 @@ int _esh_init(pmix_info_t info[], size_t ninfo)
                      *
                      * PMIX_DSTPATH has higher priority than PMIX_SERVER_TMPDIR
                      */
-                    if(_base_path != NULL)
-                        free(_base_path);
-                    _base_path = strdup((char*)info[n].value.data.ptr);
+                    dstor_tmpdir = (char*)info[n].value.data.ptr;
                     continue;
                 }
                 if (0 == strcmp(PMIX_SERVER_TMPDIR, info[n].key)) {
-                    if (NULL ==_base_path) {
-                        _base_path = strdup((char*)info[n].value.data.string);
+                    if (NULL == dstor_tmpdir) {
+                        dstor_tmpdir = (char*)info[n].value.data.string;
                     }
                     continue;
                 }
             }
         }
 
-        if (NULL ==_base_path) {
-            if (NULL == (_base_path = strdup(getenv("TMPDIR")))) {
-                if (NULL == (_base_path = strdup(getenv("TEMP")))) {
-                    if (NULL == (_base_path = strdup(getenv("TMP")))) {
-                        _base_path = strdup("/tmp");
+        if (NULL == dstor_tmpdir) {
+            if (NULL == (dstor_tmpdir = getenv("TMPDIR"))) {
+                if (NULL == (dstor_tmpdir = getenv("TEMP"))) {
+                    if (NULL == (dstor_tmpdir = getenv("TMP"))) {
+                        dstor_tmpdir = "/tmp";
                     }
                 }
             }
         }
 
-        tmp = _base_path;
-        asprintf(&_base_path, "%s/pmix_dstor_%d", tmp, getpid());
-        free(tmp);
+        asprintf(&_base_path, "%s/pmix_dstor_%d", dstor_tmpdir, getpid());
```

Comment:
Shouldn't we be checking if `_base_path` is not `NULL` before the call to `asprintf` and free'ing the old memory before reallocating it in `asprintf`? Otherwise, I would think, we would have a memory leak here if this function is called more than once.